mllib/src/test/scala/org/apache/spark/ml/recommendation/ALSSuite.scala

```diff
@@ -777,15 +778,34 @@ class ALSSuite
   }
 }
 
-class ALSCleanerSuite extends SparkFunSuite {
+class ALSCleanerSuite extends SparkFunSuite with BeforeAndAfterEach {
+
+  private var localDir: File = _
```

Comment:
I don't see why this helps. Having state in the test class is undesirable, and the tests already configure a new conf and clean up the local dir in a finally block.